aicsimageio/readers/reader.py

```diff
@@ -834,6 +834,287 @@ def mosaic_tile_dims(self) -> Optional[Dimensions]:
+        -------
+        stack: types.MetaArrayLike
+            The fully stacked array. This can be 6+ dimensions with Scene being
+            the first dimension.
+
+        """
+        mode_check = ["data", "dask_data", "xarray_data", "xarray_dask_data"]
+        if mode not in mode_check:
+            raise ValueError(
+                f"Invalid mode kwarg. Found {mode} but should be one of:"
+                f"{', '.join(mode_check)}."
+            )
+        scene_stacks = []
+        scene_names = []
+
+        if select_scenes is None:
+            select_scenes = list(range(len(self.scenes)))
+
+        for i, s in enumerate(select_scenes):
+            self.set_scene(s)
+            data = getattr(self, mode)
+
+            # Store the shape and dtype of the first scenes data
+            # to check against later scenes. If returning a DataArray
+            # store the coords and dims to use in the final output
+            if i == 0:
+                shape = data.shape
+                dtype = data.dtype
+
+                if "xarray" in mode:
+                    coords = dict(data.coords)
+                    if scene_character in coords:
+                        _ = coords.pop(scene_character)
+                    dims = data.dims
+
+            # Check other scenes against the first scene
+            else:
+                if data.shape != shape:
+                    if not drop_non_matching_scenes:
+                        raise exceptions.UnexpectedShapeError(
+                            f"All scenes must have same shape. Found shape"
+                            f"{data.shape} in scene {s} but expected"
+                            f"{shape} based on scene {select_scenes[0]}"
+                        )
+                    else:
+                        continue
+                if data.dtype != dtype:
+                    if not drop_non_matching_scenes:
+                        raise TypeError(
+                            f"All scenes must have the same dtype. Found data"
+                            f"with dtype {data.dtype} in scene {s} but expected"
+                            f"dtype {dtype} based on scene {select_scenes[0]}"
+                        )
+                    else:
+                        continue
+
+            scene_stacks.append(data)
+            scene_names.append(self.current_scene)
+
+        stack = da.stack if "dask" in mode else np.stack
+
+        if "xarray" in mode:
+            all_data = stack([x.data for x in scene_stacks])
+            return xr.DataArray(
+                all_data,
+                dims=(scene_character, *dims),
+                coords={scene_character: scene_names, **coords},
+            )
+
+        else:
+            return stack(scene_stacks)
+
+    def get_stack(
+        self,
+        drop_non_matching_scenes: bool = False,
+        select_scenes: Optional[
+            Union[List[Union[str, int]], Tuple[Union[str, int], ...]]
+        ] = None,
```

Comment:
missing the `scene_character` param.

Since these are duplicated over and over, I am tempted to allow **kwargs

Reply:
Alright I'll go with that. Thanks for all the comments and answers. Ill get to work on this stuff and hopefully I can push changes later this afternoon.